service/src/main/java/bio/terra/landingzone/stairway/flight/create/resource/step/CreateAksLogSettingsStep.java

```diff
@@ -112,4 +114,11 @@ protected String getResourceType() {
   protected Optional<String> getResourceId(FlightContext context) {
     return Optional.empty();
   }
+
+  @Override
+  public List<ResourceNameRequirements> getResourceNameRequirements() {
```

Comment:
For the future, not this PR: I feel like we need a name for things like this that aren't actually azure resources but are still _things_ we're instantiating in Azure _similar_ to resources (and subject to naming constraints etc.). (diagnostic settings are actual azure resources afaik)